src/bentoml/_internal/io_descriptors/json.py

```diff
@@ -300,10 +354,10 @@ def input_type(self) -> UnionType:
         return JSONType
 
     def openapi_schema(self) -> Schema:
-        if not self._pydantic_model:
```

Comment:
Since you are refactoring this part, maybe we can do the following
```python
if self._pydantic_model:
	return Schema(**schema.model_process_schema...)
elif self._typeddict:
	return Schema(**typed_dict_to_dict(...))
else:
	return Schema(type='object')
```